CMakeLists.txt

```diff
@@ -288,14 +294,26 @@ if(${CMAKE_SYSTEM_NAME} STREQUAL FreeBSD)
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DO_LARGEFILE=0 -Dstat64=stat -Dlstat64=lstat -Dlseek64=lseek -Doff64_t=off_t -Dfstat64=fstat -Dftruncate64=ftruncate")
 endif()
 
-option(BUILD_SHARED_LIBS "Compile shared libraries" OFF)
+option(BUILD_SHARED_LIBS "Link against dependent libraries dynamically" ON)
+option(BUILD_DYNAMIC_MODULE "Output dynamically loadable module" OFF)
```

Comment:
Why did you introduce BUILD_DYNAMIC_MODULE ?
I think you misunderstood how add_library works, calling it with SHARED/STATIC creates a shared or static library. It has *nothing* to do with the MSVC runtime versions (selected using /MD(d) or /MT(d).
By default cmake always use MD(d).
For correctness, your code should not do the following

    set_target_properties(${LIBRARY_NAME} PROPERTIES OUTPUT_NAME_DEBUG StormLibDUS)
    set_target_properties(${LIBRARY_NAME} PROPERTIES OUTPUT_NAME_RELEASE StormLibRUS)

unless you actually pass the /MT(d) flags.
Anyway most of what was done here looks wrong, so I will probably fix it after #113 is merged.